File: NFL_with_R/PassingYards_Short_vs_Long_Passes.r
pbp_r <- load_pbp(2021:2024)

pbp_r_p <-
  pbp_r |>
  filter(play_type == 'pass' & !is.na(air_yards))

pbp_r_p <-
  pbp_r_p |>
  mutate(
    pass_length_air_yards = ifelse(air_yards >= 20, "long", "short"),
    passing_yards = ifelse(is.na(passing_yards), 0, passing_yards)
  )
 
pbp_r_p |>
  filter(pass_length_air_yards == "short") |>
  pull(epa) |>
  summary()

pbp_r_p |>
  filter(pass_length_air_yards == "long") |>
  pull(epa) |>
  summary()
  

ggplot(pbp_r, aes(x = passing_yards)) +
  geom_histogram()
  
pbp_r_p |>
  filter(pass_length_air_yards == "long") |>
  ggplot(aes(passing_yards)) +
  geom_histogram(binwidth = 1) +
  ylab("Count")
  xlab("Net yards on 'long' passing plays") +
  theme_bw()

ggplot(pbp_r_p, aes(x = pass_length_air_yards, y = passing_yards)) +
  geom_boxplot() +
  theme_bw() +
  xlab("Pass length in yards(long >= 20, short < 20 yards)") +
  ylab("Yards gained (or lost) during a pass play")

File: NFL_with_Python_&_R/passing_yards_short_vs_long_passes.py
import pandas as pd
import numpy as np
import nfl_data_py as nfl
import seaborn as sns
import matplotlib.pyplot as plt
import os

pbp_py = nfl.import_pbp_data([2024])

pbp_py_p = pbp_py.query("play_type == 'pass' & air_yards.notnull()").reset_index()

pbp_py_p["pass_length_air_yards"] = np.where(
    pbp_py_p["air_yards"] >= 20, "long", "short"
)

pbp_py_p["passing_yards"] = np.where(
    pbp_py_p["passing_yards"].isnull(), 0, pbp_py_p["passing_yards"]
)
pbp_py_p["passing_yards"].describe()

pbp_py_p.query('pass_length_air_yards == "short"')["passing_yards"].describe()
pbp_py_p.query('pass_length_air_yards == "long"')["passing_yards"].describe()

pbp_py_p.query('pass_length_air_yards == "short"')["epa"].describe()
pbp_py_p.query('pass_length_air_yards == "long"')["epa"].describe()

sns.displot(data=pbp_py, x="passing_yards")
plt.show()

sns.set_theme(style="whitegrid", palette="colorblind")

pbp_py_p_short = pbp_py_p.query('pass_length_air_yards == "short"')

pbp_py_hist_short = sns.displot(data=pbp_py_p_short, binwidth=1, x="passing_yards")
pbp_py_hist_short.set_axis_labels(
    "Yards gained (or lost) during a passing play", "Count"
)
plt.show()

pass_boxplot = sns.boxplot(data=pbp_py_p, x="pass_length_air_yards", y="passing_yards")
pass_boxplot.set(
    xlabel="Pass length (long >= 20 yards, short < 20 yards)",
    ylabel="Yards gained (or lost) during a passing play",
)
plt.show()
